### PR TITLE
Fix interactive SVG example in SVG <view> doc

### PR DESCRIPTION
**Pull request by sideshowbarker**
Fixes https://github.com/mdn/content/issues/6131

---

It seems that interactive SVG features won’t work for an SVG image if the SVG markup is put directly into an HTML document — and instead, the SVG image needs to be embedded using an `object` or `iframe` element.

### CODE REVIEW

**Comment by sideshowbarker** [comment]
@mdn/core-dev I initially tried putting the SVG image into a separate file and then checking that file into the repo and having the `object` element reference the relative path to that file — but that caused this:

> `Refused to display 'https://pr6137.content.dev.mdn.mozit.cloud/en-US/docs/Web/SVG/Element/view/view.svg' in a frame because it set 'X-Frame-Options' to 'DENY'.`

File: files/en-us/web/svg/element/view/index.html [hunk 1]
```diff
@@ -66,9 +67,15 @@ <h3 id="SVG">SVG</h3>
...
+</pre>
+
 <h3 id="Result">Result</h3>
 
-<p>{{EmbedLiveSample("Example", 600, 200)}}</p>
+<p>{{EmbedLiveSample("Example", "85ch", "240px")}}</p>
```
Comment (Ryuno-Ki) [inline files/en-us/web/svg/element/view/index.html:78]: What is the motivation behind the change of length here?

File: files/en-us/web/svg/element/view/index.html [hunk 2]
```diff
@@ -66,9 +67,15 @@ <h3 id="SVG">SVG</h3>
...
...D0iNjAwIiBoZWlnaHQ9IjIwMCIgdmlld0JveD0iMCAwIDYwMCAyMDAiDQogICAgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIg0KICAgIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIj4NCiAgPGRlZnM%2BDQogICAgPHJhZGlhbEdyYWRpZW50IGlkPSJncmFkaWVudCI%2BDQogICAgICA8c3RvcCBvZmZzZXQ9IjAlIiBzdG9wLWNvbG9yPSIjOGNmZmEwIiAvPg0KICAgICAgPHN0b3Agb2Zmc2V0PSIxMDAlIiBzdG9wLWNvbG9yPSIjOGNhMGZmIiAvPg0KICAgIDwvcmFkaWFsR3JhZGllbnQ%2BDQogIDwvZGVmcz4NCg0KICA8Y2lyY2xlIHI9IjUwIiBjeD0iMTgwIiBjeT0iNTAiIHN0eWxlPSJmaWxsOnVybCgjZ3JhZGllbnQpIi8%2BDQoNCiAgPHZpZXcgaWQ9ImhhbGZTaXplVmlldyIgdmlld0JveD0iMCAwIDEyMDAgNDAwIi8%2BDQogIDx2aWV3IGlkPSJub3JtYWxTaXplVmlldyIgdmlld0JveD0iMCAwIDYwMCAyMDAiLz4NCiAgPHZpZXcgaWQ9ImRvdWJsZVNpemVWaWV3IiB2aWV3Qm94PSIwIDAgMzAwIDEwMCIvPg0KDQogIDxhIHhsaW5rOmhyZWY9IiNoYWxmU2l6ZVZpZXciPg0KICAgIDx0ZXh0IHg9IjUiIHk9IjIwIiBmb250LXNpemU9IjIwIj5oYWxmIHNpemU8L3RleHQ%2BDQogIDwvYT4NCiAgPGEgeGxpbms6aHJlZj0iI25vcm1hbFNpemVWaWV3Ij4NCiAgICA8dGV4dCB4PSI1IiB5PSI0MCIgZm9udC1zaXplPSIyMCI%2Bbm9ybWFsIHNpemU8L3RleHQ%2BDQogIDwvYT4NCiAgPGEgeGxpbms6aHJlZj0iI2RvdWJsZVNpemVWaWV3Ij4NCiAgICA8dGV4dCB4PSI1IiB5PSI2MCIgZm9udC1zaXplPSIyMCI%2BZG91YmxlIHNpemU8L3RleHQ%2BDQogIDwvYT4NCjwvc3ZnPg%3D%3D" type="image/svg+xml"&gt;&lt;/object&gt;
```
Comment (Ryuno-Ki) [inline files/en-us/web/svg/element/view/index.html:73]: I'm wondering: is it possible to use a self-closing `<object />` here?

Comment (Ryuno-Ki) [reply files/en-us/web/svg/element/view/index.html:73]: I verified, that the base64 string is decodable.

**Review by Ryuno-Ki — COMMENTED** [review]
LGTM, but I have some questions:

File: files/en-us/web/svg/element/view/index.html (hunk 1 above)
Comment (sideshowbarker) [reply files/en-us/web/svg/element/view/index.html:78]: > What is the motivation behind the change of length here?

To make the height of the iframe taller — so that the botton double-size image doesn’t get clipped you don’t get a scrollbar.

Otherwise, without increasing the height, when you click the text to make the image double-size, it exceeds the 200px height and it gets clipped at the bottom, and so then you get a scrollbar added that you need to use to scroll to see the bottom of the image.

File: files/en-us/web/svg/element/view/index.html (hunk 2 above)
Comment (sideshowbarker) [reply files/en-us/web/svg/element/view/index.html:73]: > I'm wondering: is it possible to use a self-closing `<object />` here?

Not in HTML — see https://html.spec.whatwg.org/multipage/iframe-embed-object.html#the-object-element. In text/html HTML documents, the `</object>` end tag is required (as it also is for, e.g., `script` elements in text/html documents).

File: files/en-us/web/svg/element/view/index.html (hunk 1 above)
Comment (sideshowbarker) [reply files/en-us/web/svg/element/view/index.html:78]: I should have also mentioned: I change the width to `85ch` because that is the literal value that’s used in Yari’s CSS to set the maximum width of the iframes for live samples in MDN.

But in playing around with it just now I see that actually no matter what value I set for the width in the `EmbedLiveSample` call, it seems to be getting ignored and the iframe always gets rendered with a width of 756.5px.

So now I’m not really sure what we should be setting that width value to in `EmbedLiveSample` calls. But since `85ch` is what Yari itself uses in the CSS stylesheet it serves, I guess that seems like a good choice.

**Review by Ryuno-Ki — APPROVED** [review]
Works for me!
Thanks for the fix
